payment/controllers.go

```diff
@@ -30,6 +31,126 @@ func Router(service *Service) chi.Router {
 	return r
 }
 
+// KeyRouter handles management of keys
+func KeyRouter(service *Service) chi.Router {
+	r := chi.NewRouter()
+	if os.Getenv("ENV") != "local" {
+		r.Use(middleware.SimpleTokenAuthorizedOnly)
+	}
+
+	r.Method("GET", "/{merchantId}", middleware.InstrumentHandler("GetKeys", GetKeys(service)))
+	r.Method("POST", "/", middleware.InstrumentHandler("CreateKey", CreateKey(service)))
+	r.Method("DELETE", "/{id}", middleware.InstrumentHandler("DeleteKey", DeleteKey(service)))
+	return r
+}
+
+// CreateKeyRequest includes information needed to create an order
+type CreateKeyRequest struct {
+	Merchant string `json:"merchant" valid:"-"`
+}
+
+// DeleteKeyRequest includes information needed to create an order
+type DeleteKeyRequest struct {
+	DelaySeconds int `json:"DelaySeconds" valid:"-"`
+}
+
+// CreateKey is the handler for creating keys for a merchant
+func CreateKey(service *Service) handlers.AppHandler {
+	return handlers.AppHandler(func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+		var req CreateKeyRequest
+		err := requestutils.ReadJSON(r.Body, &req)
+		if err != nil {
+			// FIXME Ask Ben what he would like us to do here instead of wrapError
+			return handlers.WrapError(err, "Error in request body", http.StatusBadRequest)
+		}
+
+		_, err = govalidator.ValidateStruct(req)
+		if err != nil {
+			return handlers.WrapValidationError(err)
+		}
+
+		encrypted, nonce := GenerateSecret()
+		// var Key Key
+		key, err := service.datastore.CreateKey(req.Merchant, encrypted, nonce)
+		if err != nil {
+			return handlers.WrapError(err, "Error create api keys", http.StatusInternalServerError)
+		}
+
+		w.WriteHeader(http.StatusOK)
+
+		if err := json.NewEncoder(w).Encode(key); err != nil {
+			return handlers.WrapError(err, "Error encoding the keys JSON", http.StatusInternalServerError)
+		}
+
+		return nil
+	})
+}
+
+// DeleteKey deletes a key
+func DeleteKey(service *Service) handlers.AppHandler {
+	return handlers.AppHandler(func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+		reqID := chi.URLParam(r, "id")
+		if reqID == "" || !govalidator.IsUUIDv4(reqID) {
+			return handlers.ValidationError(
+				"Error validating request url parameter",
+				map[string]interface{}{
+					"id": "id must be a uuidv4",
+				},
+			)
+		}
+
+		id := uuid.Must(uuid.FromString(reqID))
+
+		var req DeleteKeyRequest
+		err := requestutils.ReadJSON(r.Body, &req)
+		if err != nil {
+			// FIXME Ask Ben what he would like us to do here instead of wrapError
+			return handlers.WrapError(err, "Error in request body", http.StatusBadRequest)
```

Comment:
This is cool, as we are returning an AppHandler, we should use `handlers.WrapError` :)

Reply:
Removed here! https://github.com/brave-intl/bat-go/pull/285/commits/f5ca28286a4358dce3ee80d23a3f1c8bdc66e717